brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java

```diff
@@ -113,6 +115,7 @@ public long startTimestamp() {
     @SuppressWarnings("ClassNameSameAsAncestorName")
     private static final class HttpClientResponse extends brave.http.HttpClientResponse {
         private final RequestLog log;
+        private brave.http.HttpClientRequest request;
 
         HttpClientResponse(RequestLog log) {
```

Comment:
I was sort of expecting this to accept `HttpClientRequest` as a parameter and passed in when initializing here

https://github.com/line/armeria/blob/master/brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java#L162

I guess it does mean mutations would be present but is it a problem either way?

But anyways, reexposing, with the lazy instantiation, seems fine to me too.

Reply:
this is part of me not knowing the lifecycle. I'm not sure if the instance could be different by the time we reach response time. is that something we cam know?

Reply:
Instance is the same. In other words, the only difference between passing the originally created `HttpClientRequest` here or recreating it here is the value of `headersBuilder`, everything else should be exactly the same since it's the same `ClientRequestContext`.

Reply:
ok in that case, let's reuse it. thanks!